gateways/token_gateway.py

```diff
@@ -44,14 +56,85 @@ def get_token_metadata_from_s3(self, file: str) -> Union[TokenMetadata, None]:
         return TokenMetadata.from_dict(token_metadata)
 
     def get_token(self, id: str) -> Union[Token, None]:
-        response = self.hathor_core_client.get(TOKEN_ENDPOINT, {'id': id})
+        """Retrieve token data from hathor wallet service
+
+        :param id: token id
+        :type id: str
+        :return: token domain class created from response
+        :rtype: Union[Token, None]
+        """
+        response = self.hathor_wallet_service_client.get(TOKEN_ENDPOINT.format(id=id), {})
 
         if response is None:
             return None
 
-        token_response = HathorCoreTokenResponse.from_dict(response)
+        token_response = HathorWalletServiceTokenResponse.from_dict(response)
```

Comment:
I'm not sure whether we are just relaying the response or mapping to a well defined response type. In case we are relaying, I don't see why we would parse it.